app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+use App\Http\Requests;
+
+//additional includes
+use App\User;
+use Auth;
+use Exception;
+use Hash;
+use Response;
+use View;
+
+class UserController extends Controller
+{
+    public function login(Request $request)
+    {
+        if (Auth::attempt(['username' => $request->username, 'password' => $request->password])) {
+            return redirect()->route('user.list');
+        }
+
+        return back()->withErrors('Invalid login.');
+    }
+
+
+    public function index()
+    {
+        $formData                  = [];
+        $formData['users']         = User::select(['id', 'first_name as firstname', 'last_name as lastname', 'username'])
+                                         ->orderBy('firstname')
+                                         ->paginate(5);
```

Comment:
Please put this on a Repository file.
Just create a Repositories folder under app.
Call it like use app/Repositories/UserRepository